File: Windie_Visao/Windie-Angular/src/app/Logica/Debug.ts
export class Debug{
    private static  debugLevel : number = 4; //debug a ser exibido 0: nenhum debug, 1: erros e validações, 2: requests, 3: inputs e outputs em Json, 4: tudo

    private static logBrowser(mensagem : string) {
		console.log(mensagem);
	}
	
	public static log(mensagem: string,debugLevel: number) {
		if(debugLevel<=Debug.debugLevel) {
			Debug.logBrowser(mensagem);
		}
	}
	
	public static  logErro(mensagem: string) {
		
		Debug.log(mensagem, 1);
	}
	
	public static logFalhaNaValidacao(mensagem: string) {
		
		Debug.log(mensagem, 1);
	}
	
	
	public static  logRequest(mensagem: string) {
		
		Debug.log(mensagem, 2);
	}
	
	
	public static  logInput(mensagem: string) {
		
		Debug.log(mensagem, 3);
	}
	
	
	public static logOutput(mensagem: string) {
		
		Debug.log(mensagem, 3);
	}
	
	public static logDetalhe(mensagem: string) {
		
		Debug.log(mensagem, 4);
	}
}

File: Windie_Back/Project/src/main/java/util/Debug.java
package util;

public class Debug {
	
	public static final int debugLevel = 4; //debug a ser exibido 0: nenhum debug, 1: erros e validações, 2: requests, 3: inputs e outputs em Json, 4: tudo
	
	
	private static void log(String mensagem) {
		System.out.println(mensagem);
	}
	
	public static void log(String mensagem,int debugLevel) {
		if(debugLevel<=Debug.debugLevel) {
			Debug.log(mensagem);
		}
	}
	
	public static void logErro(String mensagem) {
		
		Debug.log(mensagem, 1);
	}
	
	public static void logFalhaNaValidacao(String mensagem) {
		
		Debug.log(mensagem, 1);
	}
	
	
	public static void logRequest(String mensagem) {
		
		Debug.log(mensagem, 2);
	}
	
	
	public static void logInput(String mensagem) {
		
		Debug.log(mensagem, 3);
	}
	
	
	public static void logOutput(String mensagem) {
		
		Debug.log(mensagem, 3);
	}
	
	public static void logDetalhe(String mensagem) {
		
		Debug.log(mensagem, 4);
	}
	
	
	


}
